hail/src/main/scala/is/hail/expr/ir/EvaluateRelationalLets.scala

```diff
@@ -0,0 +1,34 @@
+package is.hail.expr.ir
+
+import scala.collection.mutable
+
+object EvaluateRelationalLets {
+
+  def apply(ir: BaseIR): BaseIR = {
+
+    def interpretAndCoerce(value: IR): IR = Literal.coerce(value.typ, Interpret(value, optimize = false))
```

Comment:
We never want to call Interpret, do we?  Don't want CompileAndEvaluate here?

Reply:
in the case that these things will actually be used in, it'll be a TableAggregate or something, which needs to go through Interpret. 

I'm going to fix up the interpret/compileAndEvaluate stuff soon to never use interpret.

Reply:
> I'm going to fix up the interpret/compileAndEvaluate stuff soon to never use interpret.

Awesome.  All these entrypoints and code paths and recursive invocations gives me a bella ache.  Maybe move Interpret to the test module?

Reply:
yeah, that's my plan!